src/overlays/actors/ovl_Boss_Tw/z_boss_tw.c

```diff
@@ -156,183 +187,5337 @@ static ColliderCylinderInit D_8094A840 = {
+static u8 sShieldFireCharge;
+static u8 sShieldIceCharge;
+static f32 D_8094C854;
+static f32 D_8094C858;
+static u8 sTwinrovaBlastType;
+static u8 sFixedBlastType;
+static u8 sFixedBlatSeq;
+static u8 sFreezeState;
+static Vec3f sShieldHitPos;
+static s16 sShieldHitYaw;
+static u8 sBeamDivertTimer;
+static u8 D_8094C86F;
+static u8 D_8094C870;
+static s16 D_8094C872;
+static s16 D_8094C874;
+static s16 D_8094C876;
+static u8 D_8094C878;
+static s16 D_8094C87A;
+static s16 D_8094C87C;
+static u8 D_8094C87E;
+static BossTwEEffect sTWEffects[150];
+
+extern FlexSkeletonHeader D_060070E0;
+extern AnimationHeader D_06006F28;
+extern FlexSkeletonHeader D_0601F888;
+extern AnimationHeader D_0603A2D0;
+extern AnimationHeader D_06032BF8;
+extern AnimationHeader D_06035988;
+extern AnimationHeader D_060343B4;
+extern AnimationHeader D_06035030;
+extern AnimationHeader D_060338F0;
+extern AnimationHeader D_06024374;
+extern AnimationHeader D_06023750;
+extern AnimationHeader D_06022700;
+extern AnimationHeader D_06036FBC;
+extern Gfx D_0601A528[];
+extern Gfx D_0601A998[];
+extern Gfx D_0601AB00[];
+extern Gfx D_0601A430[];
+extern Gfx D_0601A790[];
+extern Gfx D_0601AA50[];
+extern Gfx D_0601A5A8[];
+extern Gfx D_0601F390[];
+extern Gfx D_0601F238[];
+extern Gfx D_0601EEB0[];
+extern Gfx D_0601EC68[];
+extern Gfx D_0601E0E0[];
+extern Gfx D_0601E020[];
+extern Gfx D_0601DDF0[];
+extern Gfx D_0601E3A0[];
+extern Gfx D_0601E2C0[];
+extern Gfx D_0601E9F0[];
+extern Gfx D_0602D940[];
+extern Gfx D_0602D890[];
+extern Gfx D_0601CEE0[];
+extern Gfx D_0601DBE8[];
+extern Gfx D_06019D40[]; // Koume Ground Crater
+extern Gfx D_06018FC0[]; // Koume Ground Smoke
+extern Gfx D_06019938[]; // koume Ground Flame
+extern Gfx D_0601BC00[];
+extern Gfx D_0601C1C0[];
+extern Gfx D_06013E98[];
+extern Gfx D_06013F98[];
+extern Gfx D_06014070[];
+extern Gfx D_06014158[];
+extern Gfx D_06013AE8[];
+extern Gfx D_06013D68[];
+extern Gfx D_06012CE0[];
+extern Gfx D_060134B8[];
+extern Gfx D_06012B38[];
+extern Gfx D_06013310[];
+extern AnimationHeader D_060216DC;
+extern AnimationHeader D_0600230C;
+extern AnimationHeader D_06001D10;
+extern AnimationHeader D_060017E0;
+extern AnimationHeader D_060012A4;
+extern AnimationHeader D_060004A4;
+extern AnimationHeader D_06000AAC;
+extern AnimationHeader D_060088C8;
+extern UNK_TYPE D_0602E170[];
+extern AnimationHeader D_06038E2C;
+extern AnimationHeader D_06032BF8;
+extern AnimationHeader D_06007CA8;
+extern AnimationHeader D_06006530;
+extern AnimationHeader D_0600578C;
+extern AnimationHeader D_06004548;
+extern AnimationHeader D_06009398;
+extern AnimationHeader D_06003614;
+extern AnimationHeader D_06003E34;
+extern AnimationHeader D_06007688;
+extern FlexSkeletonHeader D_06032020;
+extern AnimationHeader D_060244B4;
+extern Gfx D_0601F608[];
+
+void BossTw_AddDotEffect(GlobalContext* globalCtx, Vec3f* initalPos, Vec3f* initalSpeed, Vec3f* accel, f32 scale,
+                         s16 args, s16 effCnt) {
```

Comment:
see comment below about this being more of a countLimit. This number serves as a limit for how many slots it will test as being free. Since this is a shared buffer for all of the effects, trying every slot for each add attempt would result in a full buffer really quickly, so a limit is desireable